app/actions/remote/entry.ts

```diff
@@ -22,19 +24,314 @@ import {MyPreferencesRequest, fetchMyPreferences} from './preference';
 import {fetchRolesIfNeeded} from './role';
 import {ConfigAndLicenseRequest, fetchConfigAndLicense} from './systems';
 import {fetchMyTeams, fetchTeamsChannelsAndUnreadPosts, MyTeamsRequest} from './team';
+import {fetchMe, MyUserRequest} from './user';
 
 import type {Client} from '@client/rest';
+import type ClientError from '@client/rest/error';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type TeamModel from '@typings/database/models/servers/team';
 
 type AfterLoginArgs = {
     serverUrl: string;
     user: UserProfile;
     deviceToken?: string;
 }
 
+type AppEntryData = {
+    initialTeamId: string;
+    teamData: MyTeamsRequest;
+    chData: MyChannelsRequest;
+    prefData: MyPreferencesRequest;
+    meData: MyUserRequest;
+    removeTeamIds?: string[];
+    removeChannelIds?: string[];
+}
+
+type PrepareModelsArgs = {
+    operator: ServerDataOperator;
+    initialTeamId?: string;
+    removeTeams?: TeamModel[];
+    removeChannels?: ChannelModel[];
+    teamData?: MyTeamsRequest;
+    chData?: MyChannelsRequest;
+    prefData?: MyPreferencesRequest;
+    meData?: MyUserRequest;
+}
+
+const fetchAppEntryData = async (database: Database, serverUrl: string, initialTeamId: string): Promise<AppEntryData> => {
```

Comment:
why are you passing the database here? the first argument of any action is the serverUrl, with that and the DB manager you can get the actual database.